pennylane_lightning/core/src/simulators/lightning_gpu/measurements/tests/mpi/Test_StateVectorCudaMPI_Expval.cpp

```diff
@@ -0,0 +1,488 @@
+// Copyright 2022-2023 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the License);
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an AS IS BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <algorithm>
+#include <complex>
+#include <iostream>
+#include <limits>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#include <catch2/catch.hpp>
+
+#include "MPIManager.hpp"
+#include "MeasurementsGPU.hpp"
+#include "MeasurementsGPUMPI.hpp"
+#include "StateVectorCudaMPI.hpp"
+#include "StateVectorCudaManaged.hpp"
+#include "cuGateCache.hpp"
+#include "cuGates_host.hpp"
+#include "cuda_helpers.hpp"
+
+#include "TestHelpers.hpp"
+
+using namespace Pennylane::LightningGPU;
+using namespace Pennylane::Util;
+
+/// @cond DEV
+namespace {
+using namespace Pennylane::LightningGPU;
+using namespace Pennylane::LightningGPU::Measures;
+using namespace Pennylane::LightningGPU::Observables;
+using Pennylane::Util::createNonTrivialState;
+using Pennylane::Util::write_CSR_vectors;
+namespace cuUtil = Pennylane::LightningGPU::Util;
+} // namespace
+/// @endcond
+
+TEMPLATE_TEST_CASE("[Identity]", "[StateVectorCudaMPI_Expval]", float, double) {
+    using StateVectorT = StateVectorCudaMPI<TestType>;
+    const size_t num_qubits = 3;
+    auto ONE = TestType(1);
+
+    MPIManager mpi_manager(MPI_COMM_WORLD);
+    CHECK(mpi_manager.getSize() == 2);
+
+    size_t mpi_buffersize = 1;
+
+    size_t nGlobalIndexBits =
+        std::bit_width(static_cast<size_t>(mpi_manager.getSize())) - 1;
+    size_t nLocalIndexBits = num_qubits - nGlobalIndexBits;
+
+    int nDevices = 0;
+    cudaGetDeviceCount(&nDevices);
+    CHECK(nDevices >= 2);
```

Comment:
There are a couple place where we do not `CHECK` this. Is that ok?

Reply:
More added